scripts/update-core.sh.in

```diff
@@ -89,12 +89,18 @@ msg "Checking if there are critical packages to upgrade."
 if ! run_pacman -Qu ${CRITICAL_PACKAGES}; then
   msg "No updates for core packages."
 else
-  warning "You MUST restart MSYS2 to finish core packages update!!!"
-  msg "Update core packages..."
+  msg "Core packages require updating."
+  msg "Please close all other MSYS2 derived windows (e.g. terminal"
+  msg "windows, Bash sessions, etc) before proceeding."
+  read -p "Press [Enter] key when ready to start update..."
+  msg "Updating core packages..."
   if ! run_pacman -S --noconfirm --needed ${CRITICAL_PACKAGES} ${OPTIONAL_PACKAGES}; then
     error "$(gettext "'%s' failed to update core packages.")" "$PACMAN"
     exit 1 # TODO: error code
   fi
+  msg "Core packages updated."
+  warning "You MUST close this MSYS2 window now to finish !!!"
+  read -p "Press Alt-F4 or red [ X ] to close."
```

Comment:
In scripts/update-core.sh.in:
>    if ! run_pacman -S --noconfirm --needed ${CRITICAL_PACKAGES} ${OPTIONAL_PACKAGES}; then
>      error "$(gettext "'%s' failed to update core packages.")" "$PACMAN"
>      exit 1 # TODO: error code
>    fi
> +  msg "Core packages updated."
> +  warning "You MUST close this MSYS2 window now to finish !!!"
> +  read -p "Press Alt-F4 or red [ X ] to close."
Don't think this will work because after "run_pacman" no one command can't be executed. You will have fork errors instead.

I've updated the Pull Request with a fresh msg/warning arrangement, to avoid the update trap you identified.

Philip